### PR TITLE
Unambiguous char-to-number and char-to-digit conversions

### PR DESCRIPTION
**Pull request by qurbonzoda**

### CODE REVIEW

File: proposals/stdlib/char-int-conversions.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+- The correct implementation of `Char.digitToInt` relies on knowing all digit ranges among the supported
+`Char` values, see the issues [KT-30652](https://youtrack.jetbrains.com/issue/KT-30652) and [KT-39177](https://youtrack.jetbrains.com/issue/KT-39177).
+
+- Deprecating `Number.toChar()` and changing its modality will require a special support in the compiler.
+
+## Placement
+
+- module `kotlin-stdlib`
+- package `kotlin.text`
```
Comment (qurbonzoda) [inline proposals/stdlib/char-int-conversions.md:169]: `fun Char(UShort): Char`, `fun Char(Int): Char` and `Char.code` are placed in `kotlin` package

File: proposals/stdlib/char-int-conversions.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ */
+val Char.code: Int
+```
+These functions will be proposed as replacements for the deprecated conversions above.
+For example: 
+```kotlin
+char.toInt() -> char.code
+char.toShort() -> char.code.toShort()
+
+int.toChar() -> Char(int.toUShort())
```
Comment (qwwdfsad) [inline proposals/stdlib/char-int-conversions.md:105]: Worth to mention that we plan to stabilize `UShort` as well, so such migration will be stable

**Review by qwwdfsad — APPROVED** [review]
👍

File: proposals/stdlib/char-int-conversions.md [hunk 3]
```diff
@@ -136,10 +146,15 @@ to the corresponding `Char` representation.
...
+fun Int.digitToChar(radix: Int): Char
 ```
 
-`isDigit` is considered to be `true` for a `Char` if the Unicode general category of the `Char` is "Nd" (`CharCategory.DECIMAL_DIGIT_NUMBER`).
+We will also introduce an overload with no arguments having implicit `radix = 10`:
+- `fun Char.digitToInt(): Int`
+- `fun Char.digitToIntOrNull(): Int?`
+- `fun Int.digitToChar(): Char`
+
+Having overloads and not a default argument will be consistent with `String.toInt/toIntOrNull()` functions.
```
Comment (ilya-g) [inline proposals/stdlib/char-int-conversions.md:157]: Please take this paragraph from the issue KT-23451

**Review by ilya-g — APPROVED** [review]

File: proposals/stdlib/char-int-conversions.md (hunk 2 above)
Comment (qurbonzoda) [reply proposals/stdlib/char-int-conversions.md:105]: Done

File: proposals/stdlib/char-int-conversions.md (hunk 1 above)
Comment (qurbonzoda) [reply proposals/stdlib/char-int-conversions.md]: done

File: proposals/stdlib/char-int-conversions.md (hunk 3 above)
Comment (qurbonzoda) [reply proposals/stdlib/char-int-conversions.md]: done

**Pull request merged**